lib/superstore/adapters/jsonb_adapter.rb

```diff
@@ -0,0 +1,210 @@
+gem 'pg'
+require 'pg'
+
+module Superstore
+  module Adapters
+    class JsonbAdapter < AbstractAdapter
+      class QueryBuilder
+        def initialize(adapter, scope)
+          @adapter  = adapter
+          @scope    = scope
+        end
+
+        def to_query
+          [
+            "SELECT #{select_string} FROM #{@scope.klass.table_name}",
+            where_string,
+            order_string,
+            limit_string
+          ].delete_if(&:blank?) * ' '
```

Comment:
why combine string interpolation with this array-building method of generating the string? 
```
[
  "SELECT",
  select_string,
  "FROM",
  ...
]
```